File: www/app/models/project/tasks.rb
class Project::Tasks

  def initialize project
    @project = project
  end

  def create! attributes
    Task.create!(attributes.merge(project: @project))
  end

  def all
    Task.all(@project.id)
  end

  def find_by_id id
    Task.find_by_id(@project.id, id)
  end

  def find_by_name name
    Task.find_by_name(@project.id, name)
  end

  def find_by_slug slug
    Task.find_by_slug(@project.id, slug)
  end

  def new attributes={}
    Task.new(attributes.merge(project: @project))
  end

end

File: www/static/models/Tasks.js
define(function(require) {

  var
    Backbone = require('backbone'),
    Task  = require('models/Task');

  return Backbone.Collection.extend({
    model: Task,
    path: '/tasks',

    initialize: function(models, options) {
      if(options && options.project) {
        this.path = '/projects/' + options.project.id + this.path;
      }
    },

    findBySlug: function(slug){
      return this.find(function(task){
        return task.get('slug') === slug;
      });
    }

  });

});

